src/mewbot/loader.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import importlib
+import sys
+from typing import Type, Any
+
+from mewbot.core import ComponentKind
+from mewbot.component import ConfigBlock, Component, ComponentRegistry
+
+
+def load_component_from_yaml(config: ConfigBlock) -> Component:
+    required_keys = ConfigBlock.__annotations__.keys()  # pylint: disable=no-member
+
+    if set(config.keys()) != set(required_keys):
```

Comment:
Should this be != or should it be A contains B?

Reply:
Great question!
I left this as a direct equality check because it means that extra incoming properties will cause an error, making schema mistakes easier to detect.

There's a reasonable chance that there will be multiple TypedDicts representing different complex structures (such as a behaviour having triggers/conditions/actions) which is going to make this section of code more complex.

Reply:
This has been changed to require a superset, as that was necessary for Behaviour loading.